### ISSUE TITLE
Measure matched Gemma BOUNDED latency with Jev vs generated JSON

### ISSUE THREAD

**Issue by rinsakamo** [rinsakamo/relay-self#257]
## Purpose

Measure whether the merged #256 Jev/System One BOUNDED path materially reduces real RelaySelf bounded-cognition cost on the target RTX 3060 while preserving the same useful finite-choice outcome and authority invariants.

Refs #255.
Refs #256.
Refs #101.
Refs #140.
Refs #190.

## Current implementation authority

At creation:

```text
main = 264c76b35820c0b399e3b082519cc4ca4cfbf1e4
tree = 92a559defd06808cdbe303630e2c4b3d5aa6d8d9
```

Fresh current main always supersedes these hashes.

## Matched comparison

Use one Jev-capable llama.cpp server, one loaded Gemma 4 12B GGUF, one slot/context configuration, and the same RelaySelf reference BoundedChoiceRequest.

Compare:

```text
A: generated BOUNDED
   /v1/chat/completions
   strict JSON

B: Jev BOUNDED
   /v1/systemone
   logits readout
   output_tokens = 0
```

Keep THINK disabled or use only cases that resolve in BOUNDED for the direct latency comparison. A separate escalation check may confirm that explicit Jev UNRESOLVED still reaches existing THINK correctly.

## Record separately

- exact RelaySelf HEAD/tree;
- exact llama.cpp fork revision/build;
- exact GGUF path/SHA/quantization;
- GPU/driver;
- request identity and input shape;
- correctness / selected choice;
- invalid/protocol failure count;
- prompt/input tokens;
- generated/output tokens;
- per-call latency distribution (median, p95, max);
- warm-up treatment;
- server/model load excluded from per-call comparison;
- number of calls and ordering;
- Jev probabilities/confidence as observations only, not yet routing authority.

## Completion

Complete when a fresh matched run establishes one of:

- Jev materially reduces BOUNDED latency while preserving the matched outcome;
- no material latency benefit on this target;
- the current Jev path is not physically qualified, with exact failure stage/reason.

Do not canonicalize Jev, delete the generated baseline, fit a confidence threshold, or claim calibration in this transaction unless the evidence independently justifies a separate follow-up.

**Comment by rinsakamo** [comment]
Routing decision after fresh `main`/upstream review:

```text
RelaySelf main = faf479fc23504f67270134cfffd563e91f15af62
tree           = f1814ed43f027926a8368c244f63056da64f958b
open PRs       = 0
ruleset        = 23442682 active

kishida/llama.cpp jev
HEAD           = b779b6ac2ca517e83de42010e54fbe67acf1bb3c
tree           = 6c9c30cfba92bcd9be842c313082457407a0a482
```

#257 remains a **matched Jev BOUNDED latency/correctness measurement**, using the Jev fork without RelaySelf-local cache patches.

Current direct comparison stays:

```text
A. generated BOUNDED via /v1/chat/completions
B. Jev BOUNDED via /v1/systemone
```

Keep direct A/B cases resolved in BOUNDED. An explicit UNRESOLVED -> THINK smoke may verify routing, but **cached Jev -> THINK reuse is not part of #257 and must not be inferred from it**.

Reason: RelayLM #2947 already demonstrated real llama.cpp KV reuse together with a cache-on semantic/protocol failure. Cache correctness repair is now owned by `rinsakamo/relay-lm#3006`.

Deferred RelaySelf integration/qualification is tracked separately in #259. #259 will merge/consume the proven fix into a Jev-capable runtime and specifically qualify `LABEL_LOGITS -> COMPLETION` reuse.

No RelaySelf code/runtime/default mutation is required by this routing decision.

**Comment by rinsakamo** [comment]
Repository apparatus is now merged; physical #257 measurement remains unspent.

```text
main = 70d213bbb42cb36a7ebde1e3ce9ceae2fca4c30b
tree = a8131158d03ca174805aab391c1b80fcb5af6f84
PR #260 = MERGED (squash)
reviewed/tested head = 6d1ec4194abad1d8def36216bce031543f75d7e3
CI run = 35524207521
repository-contracts = PASS
pytest = PASS
lint = PASS
mineflayer-adapter = PASS
open PRs targeting main after merge = 0
```

Added canonical apparatus:

- `adapters/llama_cpp/benchmark_jev_bounded.py`
- `adapters/llama_cpp/run_jev_bounded_benchmark.sh`
- deterministic benchmark tests and adapter documentation

The harness keeps `think_allowed=false`, uses the same bounded request, alternates generated/Jev order after explicit warmups, records exact RelaySelf/Jev source/GGUF/GPU identity, raw per-call provider+wall latency/tokens/outcome/failures, and Jev probabilities/confidence as observation-only fields. It explicitly reports `cache_reuse_qualified=false`.

Current upstream Jev reference remains:

```text
kishida/llama.cpp branch jev
HEAD = b779b6ac2ca517e83de42010e54fbe67acf1bb3c
tree = 6c9c30cfba92bcd9be842c313082457407a0a482
```

Cached `Jev -> THINK` remains outside #257 and deferred to #259 / `rinsakamo/relay-lm#3006`.

Classification: `APPARATUS_READY / PHYSICAL_MEASUREMENT_UNSPENT`.

**Comment by rinsakamo** [comment]
#257 terminal result: matched Jev BOUNDED vs generated BOUNDED

Classification: `JEV_BOUNDED_MATERIAL_LATENCY_BENEFIT`

The canonical benchmark completed once on one server lifetime. Model load and server startup were excluded from per-call measurements.

## Fresh authority and runtime

```text
RelaySelf main HEAD = 6fe7520d8ad404edc7fb914636b44347323d2e06
RelaySelf main tree = 9f4b67d0f4b3e4ce7e22f927624321d1894343b4
RelaySelf checkout  = clean

kishida/llama.cpp:jev HEAD = b779b6ac2ca517e83de42010e54fbe67acf1bb3c
kishida/llama.cpp:jev tree = 6c9c30cfba92bcd9be842c313082457407a0a482
Jev checkout         = clean

llama-server = 0.4.1-dev (build 11062, commit b779b6ac2)
build        = CUDA 12.8.93, SM 86, Release, GNU 13.3
runtime      = b11062-b779b6ac2
GGUF         = /home/rinsa/models/gguf/gemma-4-12B-it-Q4_K_M.gguf
GGUF SHA256   = c088a44859de42a1966851b552ba628c0ff4419b87c4622539d69430f40024ed
GGUF type     = Q4_K - Medium; size = 7381384864 bytes
GPU           = NVIDIA GeForce RTX 3060; driver = 591.44; compute capability = 8.6
```

Preflight passed: `/health` status `ok`; `/v1/models` contained exactly the requested model; `/props` bound the model path and Jev build; `/v1/systemone` was available (invalid-body gate returned HTTP 422 before inference). Server flags were `--parallel 1 --ctx-size 4096 --gpu-layers 99 --no-cache-prompt --cache-reuse 0 --no-cache-idle-slots --reasoning off`. The request apparatus fixed `think_allowed=false`; `cache_reuse_qualified=false`.

Request ID: `mvp-flee-destination-jev-benchmark`; expected choice: `cave`.

Warmups: 2 per arm. Measured: 24 per arm. Warmup order was A,B then B,A; measured order was the canonical alternating `A,B,B,A` repeated 12 times. Valid A/B model/provider calls: 26 per arm including warmups, 52 total.

## A — generated BOUNDED, `/v1/chat/completions`

```text
measured calls       = 24
success count        = 24
provider failures    = 0
incorrect outcomes   = 0
selected / expected  = cave / cave for all 24
tokens               = prompt 449, completion 18, total 467 for all 24
finish_reason        = stop for all 24
```

Summary (raw nanoseconds):

```text
provider: median 931030165, p95 942440183, max 954165445
wall:     median 931090609, p95 942454553, max 954177642
```

Raw provider latency, ordinal 0..23 (ns):

```text
[926060385, 930388754, 931122242, 924756509, 932124253, 930471665,
 928294439, 930938088, 929324467, 942440183, 954165445, 938259681,
 936518379, 937057000, 935597630, 929432938, 933152915, 928681629,
 936550746, 928828654, 931936043, 932177854, 930439408, 930882338]
```

Raw wall latency, ordinal 0..23 (ns):

```text
[926074913, 930400216, 931132958, 924770786, 932135596, 930482004,
 928308547, 931048260, 929336604, 942454553, 954177642, 938271258,
 936529293, 937067687, 935611434, 929444555, 933166855, 928692750,
 936562699, 928839980, 931947056, 932189399, 930450404, 930897377]
```

## B — Jev BOUNDED, `/v1/systemone`

```text
measured calls       = 24
success count        = 24
provider failures    = 0
incorrect outcomes   = 0
selected / expected  = cave / cave for all 24
tokens               = prompt 612, completion 0, total 612 for all 24
finish_reason        = null (no generated text; logits readout)
```

Summary (raw nanoseconds):

```text
provider: median 479724632, p95 485675709, max 489367988
wall:     median 479736509, p95 485687948, max 489385165
```

Raw provider latency, ordinal 0..23 (ns):

```text
[475902481, 477847548, 476084413, 478887739, 478050838, 479244264,
 477180097, 480695225, 478646312, 484911018, 485636009, 489367988,
 483466779, 485675709, 478148105, 481414792, 478614161, 482454918,
 479180432, 481844283, 478652645, 482697470, 480205000, 484231843]
```

Raw wall latency, ordinal 0..23 (ns):

```text
[475914780, 477860334, 476097540, 478897882, 478060581, 479255275,
 477192817, 480706104, 478658254, 484926702, 485647239, 489385165,
 483478608, 485687948, 478159544, 481428866, 478624581, 482471318,
 479190083, 481856703, 478663008, 482706438, 480217743, 484243090]
```

Jev observations for all 24 measured calls were identical: `probabilities = {"cave": 1.0, "ridge": 0.0, "__relay_self_unresolved__": 0.0}`, `confidence = 1.0`. These fields were recorded as observations only and were not used for routing or threshold fitting.

## Bounded interpretation and boundaries

Jev/provider median latency was 48.4738% lower than generated/provider latency; wall median was 48.4759% lower. Provider p95/max and wall p95/max showed the same approximately 48.5% reduction. Both arms preserved the useful finite-choice outcome (`cave`) with normal protocol/provider completion and zero failure or wrong-choice observations. This is bounded evidence for this exact RelaySelf/Jev/GGUF/GPU transaction, not a universal routing or calibration claim.

`cache_reuse_qualified=false`. No `cache_prompt=true`, Jev -> THINK cache experiment, LABEL_LOGITS -> COMPLETION reuse, checkpoint/rollback patch, slot/cache tuning, second model, model substitution, retry, replay, reseed, fallback, confidence threshold fitting, RelaySelf production mutation, or RelayLM mutation was performed. `/slots` after the run reported `n_prompt_tokens_cache=0`.

Fresh reconciliation: open PRs targeting RelaySelf `main` = 0; ruleset `23442682` = active; #259 = OPEN; RelayLM #3006 = OPEN. Primary `/home/rinsa/relay-self` was not modified. RelaySelf/Jev exact clean clones remained clean. Repository mutations = 0.
